gnx.js

```diff
@@ -289,44 +305,81 @@ const saveObject = async function (Model, gqltype, args) {
+  const newObject = new Model(materializedModel.modelArgs)
   console.log(JSON.stringify(newObject))
   newObject.$session(session)
 
-  if (result.collectionFields) {
-    for (const collectionField in result.collectionFields) {
-      if (result.collectionFields[collectionField].added) {
-        const argTypes = gqltype.getFields()
-        const collectionGQLType = argTypes[collectionField].type.ofType
-        const connectionField = argTypes[collectionField].extensions.relation.connectionField
+  if (materializedModel.collectionFields) {
+    iterateonCollectionFields(materializeModel, gqltype, newObject._id, session)
+  }
 
-        result.collectionFields[collectionField].added.forEach(collectionItem => {
-          onSaveObject(typesDict.types[collectionGQLType.name].model, collectionGQLType, collectionItem, session, (item) => {
-            item[connectionField] = newObject._id
-          })
-        })
-      }
-      if (result.collectionFields[collectionField].updated) {
-        const argTypes = gqltype.getFields()
-        const collectionGQLType = argTypes[collectionField].type.ofType
-        const connectionField = argTypes[collectionField].extensions.relation.connectionField
-
-        result.collectionFields[collectionField].updated.forEach(collectionItem => {
-          onSaveObject(typesDict.types[collectionGQLType.name].model, collectionGQLType, collectionItem, session, (item) => {
-            item[connectionField] = newObject._id
-          })
+  return newObject.save()
+}
+
+const iterateonCollectionFields = function (materializedModel, gqltype, objectId, session) {
+  for (const collectionField in materializedModel.collectionFields) {
+    if (materializedModel.collectionFields[collectionField].added) {
+      const argTypes = gqltype.getFields()
+      const collectionGQLType = argTypes[collectionField].type.ofType
+      const connectionField = argTypes[collectionField].extensions.relation.connectionField
+
+      materializedModel.collectionFields[collectionField].added.forEach(collectionItem => {
+        onSaveObject(typesDict.types[collectionGQLType.name].model, collectionGQLType, collectionItem, session, (item) => {
+          item[connectionField] = objectId
         })
-      }
-      if (result.collectionFields[collectionField].deleted) {
-        result.collectionFields[collectionField].deleted.forEach(collectionItem => {
-          // TODO
+      })
+    }
+    if (materializedModel.collectionFields[collectionField].updated) {
+      const argTypes = gqltype.getFields()
+      const collectionGQLType = argTypes[collectionField].type.ofType
+      const connectionField = argTypes[collectionField].extensions.relation.connectionField
+
+      materializedModel.collectionFields[collectionField].updated.forEach(collectionItem => {
+        onUpdateSubject(typesDict.types[collectionGQLType.name].model, collectionGQLType, collectionItem, session, (item) => {
+          item[connectionField] = objectId
         })
```

Comment:
This code is almost the same that the one used for the added objects, we could extract this in a generic function .